scapy/utils.py

```diff
@@ -664,7 +676,104 @@ def read_all(self,count=-1):
         return plist.PacketList(res,name = os.path.basename(self.filename))
     def recv(self, size=MTU):
         return self.read_packet(size=size)
-        
+
+
+class RawPcapNgReader(RawPcapReader):
+    """A stateful pcapng reader. Each packet is returned as a
```

Comment:
According to [Wireshark Wiki](https://wiki.wireshark.org/Development/PcapNg), "PCAP Next Generation Dump File Format (or pcapng for short)" (#luckycoder ;-)).